### PR TITLE
2492: add an option to limit the number of months returned by RegionS…

### PR DESCRIPTION
**Pull request by kpethtel**
…ervice

**Story card:** 2492

## Because

RegionService is returning more data than needed for the my facilities dashboard. 

## This addresses

Adds a new optional param to the Region Service for limiting the number of months queried.

### CODE REVIEW

File: app/services/reports/region_service.rb [hunk 1]
```diff
@@ -11,11 +11,11 @@ def self.call(*args)
       new(*args).call
     end
 
-    def initialize(region:, period:, with_exclusions: false)
+    def initialize(region:, period:, with_exclusions: false, total_months_requested: MAX_MONTHS_OF_DATA)
```
Comment (rsanheim) [inline app/services/reports/region_service.rb:14]: Suggestion: a name like `number_of_months` or just `months_request` would be a bit more clear and concise

File: spec/services/reports/region_service_spec.rb [hunk 2]
```diff
@@ -277,4 +277,21 @@ def refresh_views
       expect(result[:adjusted_registrations][june_1.to_period]).to eq(5)
     end
   end
+
+  context 'without total_months_requested' do
+    it 'returns data for the default limit of 24 months ' do
+      service = Reports::RegionService.new(region: facility_group_1, period: Period.month(june_1))
+      result = service.call
+      expect(result[:period_info].count).to eq 24
```
Comment (rsanheim) [inline spec/services/reports/region_service_spec.rb:285]: 👍

**Review by rsanheim — COMMENTED** [review]
👍 Looks good - some minor feedback left.  Also some other listing to take care of there.

File: app/services/reports/region_service.rb (hunk 1 above)
Comment (kpethtel) [reply app/services/reports/region_service.rb:14]: Will do. I was feeling pretty ambivalent about the naming.

**Review by harimohanraj89 — APPROVED** [review]
Looks great to me. I'm bike-shedding a bit, but the naming still feels a bit awkward. I'd push a bit further and say even `months` is sufficient.

`RegionService.call(region: somewhere, period: current_month, months: 12)` reads very well.

Not a blocker, though. I'll let you take a final call either way and :shipit: :shipit: :shipit:

**Comment by kpethtel** [comment]
> Looks great to me. I'm bike-shedding a bit, but the naming still feels a bit awkward. I'd push a bit further and say even `months` is sufficient.
> 
> `RegionService.call(region: somewhere, period: current_month, months: 12)` reads very well.
> 
> Not a blocker, though. I'll let you take a final call either way and :shipit: :shipit: :shipit:

I haven't met a bike shed yet that couldn't be improved upon.

**Pull request merged**